### PR TITLE
chore(datasets): relax pandas req to support 2.0.0

### PR DESCRIPTION
**Pull request by deepyaman**
## Description

Support the latest and greatest 🐼 

Closes https://github.com/kedro-org/kedro/issues/2416#issuecomment-1472950837

## Development notes

For now, I've pinned `test_requirements.txt` to `pandas==2.0.0rc0` (to avoid having to `pip install --pre`), and verified that I can make modifications to get tests to pass. I'll put this on hold until pandas 2.0.0 is actually released.

I assume bandit is failing because it can't find the pandas release candidate?

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/test_requirements.txt [hunk 1]
```diff
@@ -29,8 +29,9 @@ moto==3.0.4; python_version == '3.10'
 networkx~=2.4
 opencv-python~=4.5.5.64
 openpyxl>=3.0.3, <4.0
+pandas>=1.3; python_version == '3.7'  # 1.3 for read_xml/to_xml
+pandas==2.0.0rc0; python_version > '3.7'
```
Comment (deepyaman) [inline kedro-datasets/test_requirements.txt:33]:
```suggestion
pandas>=1.3  # 1.3 for read_xml/to_xml
```

**Comment by astrojuanlu** [comment]
🔥

**Comment by astrojuanlu** [comment]
Can we relax the sqlalchemy range as well? Pandas 2.0 is compatible with SQLAlchemy 2.0

**Comment by datajoely** [comment]
Amazing work @deepyaman

**Comment by astrojuanlu** [comment]
Errors seem to come from the tests themselves

**Comment by deepyaman** [comment]
> Errors seem to come from the tests themselves

(Working on it; it's using legacy stuff to check table existence)

**Review by astrojuanlu — APPROVED** [review]
LGTM when the tests pass! 🐼

**Pull request merged**